diff --git a/spotify-tui-0.19.0/Cargo.toml b/spotify-tui-0.19.0/Cargo.toml
index 2586ea2..6aea6de 100644
--- a/spotify-tui-0.19.0/Cargo.toml
+++ b/spotify-tui-0.19.0/Cargo.toml
@@ -14,7 +14,7 @@ license = "MIT OR Apache-2.0"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-rspotify = { git = "https://github.com/Rigellute/rspotify" }
+rspotify = "0.9.0"
 tui = { version = "0.9.1", features = ["crossterm"], default-features = false }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
