diff --git a/src/model/audio.rs b/src/model/audio.rs
index 06ff832..78f6c58 100644
--- a/src/model/audio.rs
+++ b/src/model/audio.rs
@@ -49,78 +49,78 @@ pub struct AudioAnalysis {
 pub struct AudioAnalysisMeasure {
     pub start: f32,
     pub duration: f32,
-    pub confidence: f32,
+    pub confidence: Option<f32>,
 }
 
 ///[audio analysis](https://developer.spotify.com/web-api/get-audio-analysis/)
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct AudioAnalysisSection {
-    pub start: f32,
-    pub duration: f32,
-    pub confidence: f32,
-    pub loudness: f32,
-    pub tempo: f32,
-    pub tempo_confidence: f32,
-    pub key: i32,
-    pub key_confidence: f32,
-    pub mode: f32,
-    pub mode_confidence: f32,
-    pub time_signature: i32,
-    pub time_signature_confidence: f32,
+    pub start: Option<f32>,
+    pub duration: Option<f32>,
+    pub confidence: Option<f32>,
+    pub loudness: Option<f32>,
+    pub tempo: Option<f32>,
+    pub tempo_confidence: Option<f32>,
+    pub key: Option<i32>,
+    pub key_confidence: Option<f32>,
+    pub mode: Option<f32>,
+    pub mode_confidence: Option<f32>,
+    pub time_signature: Option<i32>,
+    pub time_signature_confidence: Option<f32>,
 }
 
 ///[audio analysis](https://developer.spotify.com/web-api/get-audio-analysis/)
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct AudioAnalysisMeta {
-    pub analyzer_version: String,
-    pub platform: String,
-    pub detailed_status: String,
-    pub status_code: i32,
-    pub timestamp: u64,
-    pub analysis_time: f32,
-    pub input_process: String,
+    pub analyzer_version: Option<String>,
+    pub platform: Option<String>,
+    pub detailed_status: Option<String>,
+    pub timestamp: Option<String>,
+    pub analysis_time: Option<f32>,
+    pub input_process: Option<String>,
 }
+
 ///[audio analysis](https://developer.spotify.com/web-api/get-audio-analysis/)
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct AudioAnalysisSegment {
-    pub start: f32,
-    pub duration: f32,
-    pub confidence: f32,
-    pub loudness_start: f32,
-    pub loudness_max_time: f32,
-    pub loudness_max: f32,
+    pub start: Option<f32>,
+    pub duration: Option<f32>,
+    pub confidence: Option<f32>,
+    pub loudness_start: Option<f32>,
+    pub loudness_max_time: Option<f32>,
+    pub loudness_max: Option<f32>,
     pub loudness_end: Option<f32>,
-    pub pitches: Vec<f32>,
-    pub timbre: Vec<f32>,
+    pub pitches: Option<Vec<f32>>,
+    pub timbre: Option<Vec<f32>>,
 }
 
 ///[audio analysis](https://developer.spotify.com/web-api/get-audio-analysis/)
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct AudioAnalysisTrack {
-    pub num_samples: u32,
-    pub duration: f32,
-    pub sample_md5: String,
-    pub offset_seconds: u32,
-    pub window_seconds: u32,
-    pub analysis_sample_rate: i32,
-    pub analysis_channels: u32,
-    pub end_of_fade_in: f32,
-    pub start_of_fade_out: f32,
-    pub loudness: f32,
-    pub tempo: f32,
-    pub tempo_confidence: f32,
-    pub time_signature: i32,
-    pub time_signature_confidence: f32,
-    pub key: u32,
-    pub key_confidence: f32,
-    pub mode: f32,
-    pub mode_confidence: f32,
-    pub codestring: String,
-    pub code_version: f32,
-    pub echoprintstring: String,
-    pub echoprint_version: f32,
-    pub synchstring: String,
-    pub synch_version: f32,
-    pub rhythmstring: String,
-    pub rhythm_version: f32,
+    pub num_samples: Option<String>,
+    pub duration: Option<f32>,
+    pub sample_md5: Option<String>,
+    pub offset_seconds: Option<u32>,
+    pub window_seconds: Option<u32>,
+    pub analysis_sample_rate: Option<f32>,
+    pub analysis_channels: Option<u32>,
+    pub end_of_fade_in: Option<f32>,
+    pub start_of_fade_out: Option<f32>,
+    pub loudness: Option<f32>,
+    pub tempo: Option<f32>,
+    pub tempo_confidence: Option<f32>,
+    pub time_signature: Option<i32>,
+    pub time_signature_confidence: Option<f32>,
+    pub key: Option<u32>,
+    pub key_confidence: Option<f32>,
+    pub mode: Option<f32>,
+    pub mode_confidence: Option<f32>,
+    pub codestring: Option<String>,
+    pub code_version: Option<f32>,
+    pub echoprintstring: Option<String>,
+    pub echoprint_version: Option<f32>,
+    pub synchstring: Option<String>,
+    pub synch_version: Option<f32>,
+    pub rhythmstring: Option<String>,
+    pub rhythm_version: Option<f32>,
 }
