diff --git a/tests/lockfile.rs b/tests/lockfile.rs
index 314ece9..8ddbd71 100644
--- a/tests/lockfile.rs
+++ b/tests/lockfile.rs
@@ -9,6 +9,7 @@ fn load_our_lockfile() -> Lockfile {
     Lockfile::load("Cargo.lock").unwrap()
 }
 
+/* Disabled by Gentoo
 /// Load this crate's own V2 `Cargo.lock` file
 #[test]
 fn load_our_own_v2_lockfile() {
@@ -16,6 +17,7 @@ fn load_our_own_v2_lockfile() {
     assert_eq!(lockfile.version, ResolveVersion::V2);
     assert_ne!(lockfile.packages.len(), 0);
 }
+*/
 
 /// Load example V1 `Cargo.lock` file (from the Cargo project itself)
 #[test]
